portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +25,58 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  List<String> enteries = new ArrayList<>();
+  static int count = 0;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+        String json_list = "[";
+        for(String entry : enteries){
+            json_list += entry;
+            if (count > enteries.size()){
```

Comment:
It looks like this `if` condition is not correct. `count` will start from 0, and increase up to `entries.size()-1`, which means this `if` condition will never be true, so the commas will not be properly inserted. You can confirm this by starting your devserver, inputting multiple comments (and there'll be no comments displayed immediately after you entered your second comment), and then go to your devserver's `.../data` URL. You'll see your JSON data output not comma-separated, making your Javascript JSON parse code (the `response => response.json()` in your `postComment()` function) not working as expected, causing no comments to be properly displayed.

Best approach is not to hand-construct the JSON output, but use the gson library that you added. I suppose you used the `convertToJson()` method example at [here](https://github.com/hosungs/sps/blob/master/walkthroughs/week-2-server/examples/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java#L50), but you could have just replaced the `convertJson()` method call with `convertToJsonUsingGson()` [here](https://github.com/hosungs/sps/blob/master/walkthroughs/week-2-server/examples/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java#L71). However, this all requires you to define a separate class for your comments, like the `ServerStats` class in [here](https://github.com/hosungs/sps/blob/master/walkthroughs/week-2-server/examples/server-stats/src/main/java/com/google/sps/data/ServerStats.java).

These are all too much, and I think it's better to separate all these out. The first step would be to fix this incorrect `if` condition, and it should be `count+1 < enteries.size()` (sorry for giving out a direct answer---try to think about why your original condition wouldn't work and why this new one would work). And then confirm that your web site can display multiple comments correctly. Then finally change this hand-rolled JSON construction to using gson. In order for you to retain your comment structure (having name, title, comment fields), you do need to define your own data class, but that's not really required yet, so you may just simplify a comment in your web site as just a single string, without any fields. That will allow you to continue to use the `List<String> enteries`. If you define your own data class, your `entries` list will need to change to `List<Comment> entries` (or whatever class name you pick). It's a bit advanced at this point, so I'd recommend you to just simplify your comments as just strings, not as objects.